tornadox/blocks/step_impl/first_order/ek1_projmatfree_d_nu.py

```diff
@@ -0,0 +1,147 @@
+"""Projection-matrix-free version of the EK1."""
+
+
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+
+from tornadox.blocks import sqrtutil
+
+
+@partial(jax.jit, static_argnames=("f", "df", "num_derivatives"))
+def attempt_step_forward_only(
+    *, f, df, m, c_sqrtm, p, p_inv, a, q_sqrtm, num_derivatives
+):
+    """Attempt a full step.
+
+    Extrapolate-then-correct.
+    Includes error estimation.
+    Includes a scalar, time-varying diffusion.
+
+    Assumes a dimension-derivative ordering.
```

Comment:
I assume I understand it (each dimension is it's own block), but could we make this even more explicit? Not sure how though, without making it confusing :see_no_evil:

Reply:
Probably. But (a) I hope we improve docs later anyway, and (b) this function is extremely low-level and mostly for us I guess, so the docs should not matter too much

Reply:
Well I wasn't 100% sure about the ordering so regarding (b) I'd argue that we should still make things more clear :sweat_smile: But I agree with (a) and we can fix this later.